src/storage/interpreter.rs

```diff
@@ -137,6 +137,34 @@ pub trait InterpreterStorage:
         self.storage::<ContractsState>().remove(&(contract, key))
     }
 
+    /// Fetch a range of values from a key-value mapping in a contract storage.
+    /// Returns the full range requested using optional values in case
+    /// a requested slot is unset.  
+    fn merkle_contract_state_range(
+        &self,
+        id: &ContractId,
+        start_key: &Bytes32,
+        range: Word,
+    ) -> Result<&[Option<Cow<Bytes32>>], Self::DataError>;
```

Comment:
I think it will be hard to return `&[_]`, because, in most cases, we will create a temporary object in the implementation, and we will face the problem with ownership.

Reply:
changed to vec